src/azure-cli/azure/cli/command_modules/botservice/_params.py

```diff
@@ -166,6 +166,29 @@ def load_arguments(self, _):
         c.argument('enable_enhanced_auth', help='If true, enables enhanced authentication features. Must be true for --trusted-origins parameters to work.', arg_type=get_three_state_flag())
         c.argument('trusted_origins', nargs='+', help='Space separated Trusted Origins URLs (must use HTTPS) e.g. --trusted-origins https://mybotsite1.azurewebsites.net https://mybotsite2.azurewebsites.net')
 
+    with self.argument_context('bot directline site create') as c:
+        c.argument('enable_enhanced_auth', help='If true, enables enhanced authentication features. Must be true for --trusted-origins parameters to work.', arg_type=get_three_state_flag())
+        c.argument('is_enabled', options_list=['--enabled'], arg_type=get_three_state_flag(), help='Add the Direct Line Site in an enabled state.')
+        c.argument('support_v3', help='If true, v3 protocol will be enabled on the channel.', arg_type=get_three_state_flag())
+        c.argument('show_secrets', options_list=['--with-secrets'], help='Show secrets in response for the channel.', arg_type=get_three_state_flag())
+        c.argument('site_name', options_list=['-s', '--site-name'], help='Name of the Directline channel site.')
+        c.argument('trusted_origins', nargs='+', help='Space separated Trusted Origins URLs (must use HTTPS) e.g. --trusted-origins https://mybotsite1.azurewebsites.net https://mybotsite2.azurewebsites.net')
+
+    with self.argument_context('bot directline site delete') as c:
+        c.argument('site_name', options_list=['-s', '--site-name'], help='Name of the Directline channel site.')
```

Comment:
Can we consider abstracting site_name to a name type using CLIArgumentType? What about ['-n', '--name'] for options_list?

Reply:
Thanks for the comment! We require the name of the bot that owns the directline site sub-resource, which we accept via `['-n', '--name']`. We use these options throughout the command group (az bot).

With that in mind, I think this should continue with `['-s', '--site-name']`, but if there are other suggestions for the parameter I'm happy to discuss it further  😄